client/state/country-states/reducer.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	COUNTRY_STATES_RECEIVE,
+	COUNTRY_STATES_REQUEST,
+	COUNTRY_STATES_REQUEST_FAILURE,
+} from 'state/action-types';
+import { itemSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+// Stores the complete list of states, indexed by locale key
+export const items = createReducer( {}, {
+	[ COUNTRY_STATES_RECEIVE ]: ( state, action ) => ( { ...state, [ action.countryCode ]: action.countryStates } ),
+}, itemSchema );
+
+// Tracks states list fetching state
+export const isFetching = createReducer( {}, {
+	[ COUNTRY_STATES_REQUEST ]: ( state, { countryCode } ) => ( { ...state, [ countryCode ]: true } ),
+	[ COUNTRY_STATES_RECEIVE ]: ( state, { countryCode } ) => ( { ...state, [ countryCode ]: false } ),
```

Comment:
Thanks for bringing that up, I noticed how there seem to be different conventions in `action-types.js`:

```js
*_FETCH
*_FETCH_COMPLETED
*_FETCH_FAILED

*_REQUEST
*_REQUEST_FAILURE
*_REQUEST_SUCCESS

*_RECEIVE
*_REQUEST
*_REQUEST_FAILURE
```

Do they serve different purposes?
Just to clarify: You suggested `COUNTRY_STATES_REQUEST_SUCCESS` because we get the data from an API request and not from a different source?